components/frontend-react/src/sagas/userSaga.js

```diff
@@ -0,0 +1,46 @@
+import { takeLatest, put, call, take } from 'redux-saga/effects';
+import { userActions, downloadActions } from '../constants';
+import { verifyUserCredentials, addNewUser, getSignupRequests, approveSignupRequest, getHeavyUsers } from '../API/Users';
+import { fetchDownloads,  } from '../API/Downloads';
+
+function* userLoginWorkerSaga (action) {
+  const response = yield call(verifyUserCredentials, action);
+  sessionStorage.setItem('token', response.headers.token);
```

Comment:
What is token is not there in `response`?

Reply:
I could not understand the question Chathu. Can you please explain it to me?

Reply:
@singhrohan62 Sorry if I confuse you. What if `response.headers.token` is undefined?

Reply:
Chathu, I had thought about this, and came up with an idea that when we receive token from response, we can issue another GET request in order to check whether token is valid or not. Do you think it's a good idea ?

Reply:
No, as I remember, the tokens are always generated with each request, like a rolling token. @vivonk  correct me if I'm wrong.

 And making API call for each request is also not good. Just put validation to check if `response` or `response.headers` is there. If they ever are `undefined`, that will cause an exception and the app might crash.

@vivonk